oximeter/oximeter/src/quantile.rs

```diff
@@ -0,0 +1,393 @@
+                // equally valid estimates,  by convention we take their average.
+                return Ok(0.5 * self.marker_hghts[index]
+                    + 0.5 * self.marker_hghts[index + 1]);
+            }
+        }
+        index = index.max(0.);
+        let mut index = index.round_ties_even() as usize;
+        index = min(index, len - 1);
+        Ok(self.marker_hghts[index])
+    }
+
+    /// Return the sample size.
+    pub fn len(&self) -> u64 {
+        self.marker_pos[4] as u64
+    }
+
+    /// Determine whether the sample is empty.
+    pub fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+
+    /// Append a value/observation to the population and adjust the heights.
+    pub fn append<T>(&mut self, value: T) -> Result<(), QuantileError>
+    where
+        T: HistogramSupport,
+    {
+        if !value.is_finite() {
+            return Err(QuantileError::NonFiniteValue(format!("{:?}", value)));
+        }
+
+        // we've already checked that the value is finite.
+        let value = value.to_f64().unwrap();
+
+        // n[4] is the sample size.
+        if self.marker_pos[4] < 5 {
+            self.marker_hghts[self.marker_pos[4] as usize] = value;
+            self.marker_pos[4] += 1;
+            if self.marker_pos[4] == 5 {
+                float_ord::sort(&mut self.marker_hghts);
+            }
+            return Ok(());
+        }
+
+        // Find cell k.
+        let mut k: usize;
+        if value < self.marker_hghts[0] {
+            self.marker_hghts[0] = value;
+            k = 0;
+        } else {
+            k = 4;
+            for i in 1..5 {
+                if value < self.marker_hghts[i] {
+                    k = i;
+                    break;
+                }
+            }
+            if self.marker_hghts[4] < value {
+                self.marker_hghts[4] = value;
+            }
+        };
+
+        // Increment all positions greater than k.
+        for i in k..5 {
+            self.marker_pos[i] += 1;
+        }
+        for i in 0..5 {
+            self.desired_marker_pos[i] += self.desired_marker_incrs[i];
+        }
+
+        // Adjust height of markers.
+        for i in 1..4 {
+            // unwrap is safe because we know the exact marker positions as literals.
+            let d = self.desired_marker_pos[i]
+                - self.marker_pos[i].to_f64().unwrap();
+            if d >= 1. && self.marker_pos[i + 1] - self.marker_pos[i] > 1
+                || d <= -1. && self.marker_pos[i - 1] - self.marker_pos[i] < -1
+            {
+                let d = Float::signum(d);
+                let q_new = self.parabolic(i, d);
+                if self.marker_hghts[i - 1] < q_new
+                    && q_new < self.marker_hghts[i + 1]
+                {
+                    self.marker_hghts[i] = q_new;
+                } else {
+                    self.marker_hghts[i] = self.linear(i, d);
+                }
+                let delta = d.round_ties_even() as i64;
```

Comment:
Same note here, why `round_ties_even()` as opposed to `round()`? Actually, isn't `delta == d` by construction, since `d = d.signum()`? Why round at all?

Reply:
Yep. I don't think it's strictly necessary, I'll update it.